meta_model_building/get_opt_params.py

```diff
@@ -6,62 +6,200 @@
     'train_size': 0.75,
-    'test_size': 0.25,
-    'generations': 5,
-    'population_size': 20,
-    'verbosity': 3}
+    'test_size': 0.25
+    }
 
-housing = load_boston()
-x_train, x_test, y_train, y_test = train_test_split(data, target,
-                                                    train_size=run_param['train_size'], 
-                                                    test_size=run_param['test_size'],
-                                                    random_state = seed['split_seed'])
+run_param = {
+    'population_size': population,
+    'verbosity': verbosity,
+    'generations': generations_per_epoch,
+    'random_state': seed['tpot_seed'],
+    'config_dict': config_dict,
+    'warm_start': True
+    }
 
-tpot = TPOTRegressor(generations=run_param['generations'], population_size=run_param['population_size'],
-                    verbosity=run_param['verbosity'], random_state = seed['tpot_seed'])
+x_train, x_test, y_train, y_test = train_test_split(
+    data, target, train_size=split_param['train_size'],
+    test_size=split_param['test_size'], random_state=seed['split_seed']
+    )
 
-tpot.fit(x_train, y_train)
-print(tpot.score(x_test, y_test))
+tpot = TPOTRegressor(**run_param)
+train_scores = []
+test_scores = []
+best_pipelines = []
 
-time = time.time()
-output_name = input_file_name.split(".")[0] + '-' + str(time)
+start_time = time.time()
+time_int = int(round(start_time))
+output_name = input_file_name.split(".")[0] + '-' + str(time_int)
 output_python = output_name + '.py'
 output_pickle = output_name + '.pickle'
 
-tpot.export(output_python)
+for i_gen in range(epochs):
+    print(i_gen)
+    tpot.fit(x_train, y_train)
+    train_scores.append(tpot.score(x_train, y_train))
+    test_scores.append(tpot.score(x_test, y_test))
+    best_pipelines.append(tpot._optimized_pipeline)
+    tpot.export(output_name + '-' + str(i_gen) + '.py')
+print('train:', train_scores)
+print('test:', test_scores)
+
+
+finish_time = time.time()
+
 
 # Prepare metadata dictionary for pickling
-pickle_dict = tpot.evaluated_individuals_
+pickle_dict = {}
+pickle_dict['evaluated_individuals'] = tpot.evaluated_individuals_
 pickle_dict.update(run_param)
+pickle_dict.update(split_param)
 pickle_dict.update(seed)
-pickle_dict['time'] = time
+pickle_dict['test_scores'] = test_scores
+pickle_dict['train_scores'] = train_scores
+
+pickle_dict['model_space'] = model_space
+pickle_dict['preprocessor'] = preprocessor
+pickle_dict['epochs'] = epochs
+pickle_dict['generations_per_epoch'] = generations_per_epoch
+pickle_dict['population'] = population
+pickle_dict['seed_value'] = seed_value
+pickle_dict['conifg_dict'] = config_dict
+
+pickle_dict['duration'] = finish_time - start_time
 pickle_dict['output_pickle'] = output_pickle
-pickle_dict['output_python'] = output_python
 
-with open(output_pickle, 'w') as pickle_file:
-    pickle.dump(pickle_dict, pickle_file)
\ No newline at end of file
+with open(output_pickle, 'wb') as pickle_file:
```

Comment:
I thought this was the right thing to do, but I didn't mention it last time because I wasn't sure. I see you've figured it out!